### PR TITLE
Replacing git reference from master to version tags

### PR DESCRIPTION
**Pull request by gaiksaya**
Used outputs as it will be easy to replace/modify just one code block in future

### CODE REVIEW

File: .github/workflows/test-build-deb.yml [hunk 1]
```diff
@@ -108,7 +108,20 @@ jobs:
...
+      tag: ${{steps.version.outputs.p_tag}}
     steps:
+      - uses: actions/checkout@v1
+      
+      - name: Getting OD version
+        id: version
+        run: |
+          cd elasticsearch/bin
+          OD_VERSION=`./version-info --od`
+          echo "::set-env name=od_version::$OD_VERSION"
```
Comment (peterzhuamazon) [inline .github/workflows/test-build-deb.yml:121]: I think we dont need to set-env for od_version as it is not used anywhere.

File: .github/workflows/test-build-deb.yml [hunk 2]
```diff
@@ -218,14 +231,17 @@ jobs:
         java: [14]
     steps:
       - uses: actions/checkout@v1
+
       - uses: actions/checkout@v1
```
Comment (peterzhuamazon) [inline .github/workflows/test-build-deb.yml:235]: Two checkout?

Comment (peterzhuamazon) [reply .github/workflows/test-build-deb.yml:235]: Same thing 174-176

Comment (gaiksaya) [reply .github/workflows/test-build-deb.yml:235]: Was already present! Didn't touch the existing code. though one checkouts the workflow repo and other the plugin repo. Not sure why we need both.

Comment (peterzhuamazon) [reply .github/workflows/test-build-deb.yml:234]: Didnt realize it checkout 2 repos. This one resolved.

**Pull request merged**